CONTRIBUTING.rst

```diff
@@ -106,12 +106,65 @@ Before you submit a pull request, check that it meets these guidelines:
+To follow the principle of avoiding the dependency on other functionalities in case of objects, **stubs** are used. Stubs mimic the actual objects, but have minimum functionality and *fixed, well defined properties* used for assertions.
+
+It requires a little more effort to implement stubs of the pyfar classes. Therefore, stub utilities are provided in *pyfar/testing/stub_utils.py* and imported in *confest.py*, where the actual stubs are implemented.
+
+- Note: the stub utilities are not meant to be imported to test files directly or used for other purposes than testing. They solely provide functionality to create fixtures.
+- The utilities simplify and harmonize testing within the pyfar package and improve the readability and reliability.
+- The implementation as the private submodule ``pyfar.testing.stub_utils``  further allows the use of similar stubs in related packages with pyfar dependency (e.g. other packages from the pyfar family).
+To get an idea of the recommended stub workflow have a look at the ``sine`` fixure in *conftest.py*.
+
+**Pyfar Stubs as Dummies**
+
+Beside the use of stubs replacing objects, it is highly recommended to use **stubs as dummies**. Dummies could provide some data or several related variables needed to call a certain function (i.e. time data and sampling rate), while the actual values are of no importance.
+
+A good example is ´´test_signal_init´´ in *test_signal.py*.
+
+**When Not to Use Stubs**
+
+Sometimes, the dependency on another pyfar functionality is desired, so a stub makes no sense. Nevertheless, consider using a fixture, as for example done with the ``sine_signal`` fixture in *conftest.py*.
+
+**Mocks**
+
+Mocks are similar to stubs but used for **behavioral verification**. For example, a mock can replace a function or an object to check if it is called with correct parameters. A main motivation for using mocks is to avoid complex or time-consuming external dependencies, for example database queries.
+
+- A typical use case of mocks in the pyfar context is hardware communication, for example reading and writing of large files or audio in- and output. These use cases are rare compared to tests performing state verification with stubs.
+- In contrast to some other guidelines on mocks, external depencies do *not* need to be mocked in general. Failing tests due to changes in external packages are meaningful hints to modify the code.
+- Examples of internal mocking can be found in *test_io.py*, indicated by the pytest ``@patch`` calls.
+
+Pytest Tips
+~~~~~~~~~~~
+Pytest provides several, sophisticated functionalities which could reduce the effort of implementing tests.
+
+- Similar tests executing the same code with different variables can be `parametrized <https://docs.pytest.org/en/stable/example/parametrize.html>`_. An example is ``test___eq___differInPoints`` in *test_coordinates.py*.
```

Comment:
I would suggest to briefly mention coverage with an example, e.g.

`pytest --cov=. --cov-report=html`